mlflow/gluon.py

```diff
@@ -0,0 +1,184 @@
+import os
+
+import gorilla
+import mxnet
+from mxnet import gluon
+from mxnet import sym
+from mxnet.gluon import HybridBlock
+from mxnet.gluon.contrib.estimator import Estimator, TrainEnd, EpochEnd
+from mxnet.gluon.nn import HybridSequential
+
+import mlflow
+from mlflow.exceptions import MlflowException
+from mlflow.models import Model
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils import experimental
+from mlflow.utils.autologging_utils import try_mlflow_log
+
+FLAVOR_NAME = "gluon"
+_MODEL_SAVE_PATH = "net"
+
+
+@experimental
+def load_model(model_uri):
+    """
+    Load a Gluon model from a local file or a run.
+
+    :param model_uri: The location, in URI format, of the MLflow model. For example:
+
+                      - ``/Users/me/path/to/local/model``
+                      - ``relative/path/to/local/model``
+                      - ``s3://my_bucket/path/to/model``
+                      - ``runs:/<mlflow_run_id>/run-relative/path/to/model``
+
+                      For more information about supported URI schemes, see
+                      `Referencing Artifacts <https://www.mlflow.org/docs/latest/tracking.html#
+                      artifact-locations>`_.
+
+    :return: A Gluon model instance.
+
+    >>> # Load persisted model as a Gluon model, make inferences against an NDArray
+    >>> model = mlflow.gluon.load_model("runs:/" + gluon_random_data_run.info.run_id + "/model")
+    >>> model(nd.array(np.random.rand(1000, 1, 32)))
+    """
+    local_model_path = _download_artifact_from_uri(artifact_uri=model_uri)
+
+    model_arch_path = os.path.join(local_model_path, "data", _MODEL_SAVE_PATH) + "-symbol.json"
+    model_params_path = os.path.join(local_model_path, "data", _MODEL_SAVE_PATH) + "-0000.params"
+    ctx = mxnet.cpu()
```

Comment:
Not knowing much about gluon -- is it common to load using cpu instead of gpu? Should this be an option to the load_model?

Reply:
Yes, it should be an option. I'll add that.